File: tests/parsers/samples/code/js/simple.js
import { ComplexAnalyzer, buildDefaultAnalyzer, describeMetrics } from "./complex.js";

export function simpleFlow() {
    const analyzer = buildDefaultAnalyzer();
    const metrics = describeMetrics([1, 2, 3]);
    metrics.baseline = analyzer.distance(10, 3);
    return metrics;
}

export function summarize() {
    const result = simpleFlow();
    return `baseline=${result.baseline}, rms=${result.rms.toFixed(2)}`;
}

File: tests/parsers/samples/code/py/simple.py
"""Simple workflow that imports symbols from complex.py."""

from complex import build_default_analyzer, describe_metrics


def simple_flow() -> dict[str, float]:
    analyzer = build_default_analyzer()
    baseline = analyzer.distance(10, 3)
    metrics = describe_metrics([1.0, 2.0, 3.0])
    metrics["baseline"] = baseline
    return metrics


def summarize() -> str:
    result = simple_flow()
    return f"Summary: baseline={result['baseline']}, rms={result['rms']:.2f}"
